pkg/handler/handler_test.go

```diff
@@ -200,3 +200,32 @@ func TestToProtoBinary(t *testing.T) {
 	}
 	// TODO: add some kind of roundtrip test here
 }
+
+func TestConcurrentReadStaleCache(t *testing.T) {
+	var s spec.Swagger
+	err := s.UnmarshalJSON(returnedSwagger)
+	if err != nil {
+		t.Errorf("Unexpected error in unmarshalling SwaggerJSON: %v", err)
+	}
+
+	mux := http.NewServeMux()
+	o := NewOpenAPIService(&s)
+	if err := o.RegisterOpenAPIVersionedService("/openapi/v2", mux); err != nil {
+		t.Errorf("Unexpected error in register OpenAPI versioned service: %v", err)
+	}
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	returnedJSON := string(normalizeSwaggerOrDie(returnedSwagger))
+
+	results := make(chan string)
+	for i := 0; i < 2; i++ {
+		go func() { results <- getBodyOrDie(server) }()
```

Comment:
I know the code doesn't let you do that easily, but should we try to get different types of content at the same time, i.e. proto and json for example?